doc/compiler/optimizer/DataFlowEngine.md

```diff
@@ -127,4 +127,61 @@ In general using structures as our program fragments enables us to save roughly
+The value of `x+y` can be stored in a temporary which can be used in later blocks instead of computing `x+y` again.
+
+[TR_LocalTransparency](https://github.com/eclipse/omr/blob/f2bc4f8f6eb09f6cc8fc4ba48717de4880b970e3/compiler/optimizer/LocalAnalysis.hpp#L230)
+runs before [TR_LocalAnticipatability](https://github.com/eclipse/omr/blob/f2bc4f8f6eb09f6cc8fc4ba48717de4880b970e3/compiler/optimizer/LocalAnalysis.hpp#L286):
+- `TR_LocalTransparency::_transparencyInfo[symRefNum]`: A bit vector indicates which expressions (identified by local index) will not be
+affected by the sym ref if the sym ref is changed in the block
+- `TR_LocalTransparency::_info[blockNum]._analysisInfo`: A bit vector indicates whether or not the expression (identified by local index)
+is transparent to the block
+
+### Local Anticipatability
+
+Local Transparency does not look at whether or not an expression is evaluated in a certain block. It looks at an expression regardless
+of where it is computed. Local Anticipatablility looks at the expressions that are computed in the block. If an expression is locally
+anticipatable in a block, the expression is computed in the block and can be moved out of the block without changing the value of the
+expression. For exmaple, if `x+y` is computed in `block_10`, the expression `x+y` can be a candidate of being locally anticipatable in
+`block_10`. If `x+y` is not computed in `block_10`, the expression `x+y` will not be in the Local Anticipatablility set.
+
+[TR_LocalAnticipatability](https://github.com/eclipse/omr/blob/f2bc4f8f6eb09f6cc8fc4ba48717de4880b970e3/compiler/optimizer/LocalAnalysis.hpp#L286)
+runs after [TR_LocalTransparency](https://github.com/eclipse/omr/blob/f2bc4f8f6eb09f6cc8fc4ba48717de4880b970e3/compiler/optimizer/LocalAnalysis.hpp#L230)
+- `TR_LocalAnticipatability::_info[blockNum]._analysisInfo`: A bit vector indicates whether or not the expression (identified by local
+index) that is being computed in the block is locally anticipatable
+- `TR_LocalAnticipatability::_info[blockNum]._downwardExposedAnalysisInfo`: A bit vector indicates whether or not the value of an
+expression (identified by local index) would be changed if it were computed at the very end of the block
+- `TR_LocalAnticipatability::_info[blockNum]._downwardExposedStoreAnalysisInfo`: A bit vector indicates whether or not the value of a
+store expression (identified by local index) would be changed if it were computed at the very end of the block
+
+### Global Anticipatability
+
+Global Anticipatablility looks at if an expression can be moved acorss CFG as a whole. For example, `x+y` is globally anticipatable if
+`x+y` is computed in `block_10` and `block_11` and it can be moved out of both blocks. Global Anticipatablility gives a set of optimal
+points where the computation can be done. If an expression is not globally anticipatable in the block, this block is not an optimal
```

Comment:
I'm not sure the definition of global anticipatability is right. It represents a few things if an expression is globally anticipatable at a certain point
1. It is evaluated (i.e. encountered or computed) on all paths following that point, meaning adding the expression at that point won't make anything worse 
2. If evaluated at that point, the value of the expression would be the same as the next time the expression is encountered on all paths following that point